### DISCUSSION TITLE
Format of results_data

### DISCUSSION THREAD

**Discussion by crcrewso** [jrkerns/pylinac#352] [Ideas]
For each test type's results_data() method, I think a format needs to be established, I'd rather propose the structures here and discuss them before I start creating the code and PR's 

Currently for Winston-Lutz it's a human readable name similar to the code variable definition, without the underscores, and with no depth

For CatPhan analysis, my proposed format is a bit more complex 

Basic Format ` results_data[<module>][<result type or sub module>] `

I.e. ` results_data['CTP404']['HUs']['Air'] `

This structure has one disadvantage, the depth (or dimensionality) of the structure is ambiguous and relies on strong documentation. Additionally, should anything that possibly has more than one analysis method or phantom type have a parameters tag

```
results_data['method'] = CBCT, WL, etc
results_data['phantom'] = CatPhan604, CatPhan504, etc.
```

**Comment by jrkerns** [comment]
Yeah, this gets sticky quickly. Especially since the IDE's introspection may not introspect the dict keys, meaning the user has to read the documentation to find the keys/remind them the key names. While I've already got a start on this as a dict, I'm thinking of using a namedtuple or dataclass, as these can be introspected easier.
